apache2/msc_pcre.c

```diff
@@ -152,19 +148,15 @@ void *msc_pregcomp_ex(apr_pool_t *pool, const char *pattern, int options,
 
     #ifdef WITH_PCRE_STUDY
         #ifdef WITH_PCRE_JIT
-                pe = pcre_study(regex->re, PCRE_STUDY_JIT_COMPILE, &errptr);
+                pe = pcre_study(regex->re, PCRE_STUDY_EXTRA_NEEDED|PCRE_STUDY_JIT_COMPILE, &errptr);
```

Comment:
I'm a little ashamed to admit it, but I couldn't find this `PCRE_STUDY_EXTRA_NEEDED` flag at all. What does it do? Could you add some note to description of this PR?

Isn't it necessary in case of pcre2?

Reply:
It's to have pcre_study() creating the extra data instead of doing in afterwards.
As PCRE1 is almost obsolete anyway, I'll revert this change that's not really mandatory.
Maybe we should challenge PCRE1 support also ...

Reply:
Perhaps we should bump the default PCRE engine to v2 (in a separated PR).